lib/package-managers/npm.js

```diff
@@ -9,21 +10,37 @@ const packageJson  = require('package-json');
 const ProxyAgent   = require('https-proxy-agent');
 const npmConf      = require('npm-conf')();
 const proxyUrl     = require('get-proxy')();
+const registryUrl  = require('registry-url');
 
-let agent;
+let proxyAgent;
+let noProxy;
 if (proxyUrl) {
     const proxy = new URL(proxyUrl);
-    agent = new ProxyAgent({
+    proxyAgent = new ProxyAgent({
         host: proxy.hostname,
         port: proxy.port,
         secureProxy: proxy.protocol === 'https:',
         keepAlive: true,
-        maxSockets: npmConf.get('maxsockets') || 50, // https://docs.npmjs.com/misc/config#maxsockets
+        maxSockets: npmConf.get('maxsockets'),
         ca: npmConf.get('ca'),
         key: npmConf.get('key'),
         cert: npmConf.get('cert'),
         rejectUnauthorized: npmConf.get('strict-ssl')
     });
+
+    noProxy = process.env.NO_PROXY || process.env.no_PROXY || process.env.no_proxy;
+    if (noProxy === '*') {
+        noProxy = true;
+    } else if (noProxy) {
+        noProxy = noProxy.split(/\s*,\s*/).map(np => {
```

Comment:
Can you add a comment clarifying this somewhat arcane block of code? Also, I am a little confused about `noProxy`'s type. Is it a Boolean or something else? I would re-using the same variable for different typed values.

Reply:
Added comments and tried to clear things up a bit by renaming and not re-using. Let me know if there's still something unclear.